File: EmailAuth/app/src/main/java/com/shankaryadav/www/emailauthenticationtesting/MainActivity.java
package com.shankaryadav.www.emailauthenticationtesting;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class MainActivity extends AppCompatActivity {

    TextView textView;

    Button button;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate (savedInstanceState);
        setContentView (R.layout.activity_main);

        textView = findViewById (R.id.textview);
        button = findViewById (R.id.logout);

        FirebaseUser currentUser = FirebaseAuth.getInstance ().getCurrentUser ();
        if (currentUser != null){
            textView.setText (currentUser.getEmail ().toString ());
        }


    }
}

File: TicTacToe/app/src/main/java/com/www/shankaryadav/tictactoe/MainActivity.kt
package com.www.shankaryadav.tictactoe

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.design.snackbar
import org.jetbrains.anko.toast

class MainActivity : AppCompatActivity() {



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button.setOnClickListener {
            val intent = Intent(this, ScannerActivity::class.java)
            startActivityForResult(intent,1)
        }

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        if (data == null) {
            return
        }

        textView.text = data.getStringExtra("BRCode")
    }
}
